src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -843,6 +886,9 @@ public void onResponse(RCFResultResponse response) {
         @Override
         public void onFailure(Exception e) {
             try {
+                if (e instanceof CompletionException) {
+                    e = new ResourceNotFoundException(e.getMessage());
+                }
```

Comment:
Why are we adding an extra check here?

Reply:
Originally the exception returned by the transport service call to the `RCFResultAction` when we attempt to retrieve and process a model from the checkpoint index is a `ResourceNotFoundException` thrown by the `ModelManager` . Since we're now using the sdkClient to execute the request, the returned exception is of type `CompletionException`. Since the `ResourceNotFoundException` is used to trigger cold start, I have opted to convert this exception back to a `ResourceNotFoundException`